src/ve-utils/mms-api-client/Element.service.ts

```diff
@@ -118,51 +118,52 @@ export class ElementService {
             this.httpSvc.ping(key, weight)
             return this.inProgressElements[key] as VePromise<T>
         }
-        const deferred = this.$q.defer<T>()
-        const cached: T = this.cacheSvc.get<T>(requestCacheKey)
-        if (cached && !refresh) {
-            deferred.resolve(cached)
-            return deferred.promise
-        }
-        const deletedRequestCacheKey = this.getElementKey(reqOb, reqOb.elementId)
-        deletedRequestCacheKey.push('deleted')
-        const deleted = this.cacheSvc.get<ElementObject>(deletedRequestCacheKey)
-        if (deleted) {
-            deferred.reject({
-                status: 410,
-                data: { recentVersionOfElement: deleted },
-                message: 'Deleted',
+        this._addInProgress(
+            key,
+            new this.$q<T>((resolve, reject) => {
+                const cached: T = this.cacheSvc.get<T>(requestCacheKey)
+                if (cached && !refresh) {
+                    return resolve(cached)
```

Comment:
this applies to all refactored functions in this and project service - if it's in the cache, need to delete the in progress key before returning (since the previous code returns before setting inProgress

Reply:
Roger